### DISCUSSION TITLE
WHEN THEN ELSE?

### DISCUSSION THREAD

**Discussion by harishm11** [hyperjumptech/grule-rule-engine#368] [Q&A]
Does this rule engine support WHEN THEN ELSE? i would like to execute something else when "When" condition is not satisfied. I feel if its part of the rule instead of code, it will be better programming. 

Please advise.

Tried below but getting syntax error...

rule "example_rule"
when
    Customer.Age > 25 && Customer.Age < 60
then
    Discount = 20;
    Multicar = "Discount";
else
    Discount = 10;
    Multicar = "No Discount";
end

**Comment by newm4n** [comment]
In rule engine, you have to think like:

```
RuleA
   when 
      conditionaA == true
   then
      do something
   else
      automatically look the other rules

RuleB
   when 
      conditionaB == true
   then
      do something
   else
      automatically look the other rules

RuleC
   when 
      conditionaC == true
   then
      do something
   else
      automatically look the other rules
```

If `conditionA`, `conditionB` and `conditionC` are all `true` the engine will see a conflict, it will then try it best to choose a rule to execute. You may give a hint to the engine to select which rule to execute upon conflict like `salience`.
